File: tests/expected/coverage.cpp
#include "py14/runtime/builtins.h" // NOLINT(build/include_order)
#include "py14/runtime/range.hpp"  // NOLINT(build/include_order)
#include "py14/runtime/sys.h"      // NOLINT(build/include_order)
#include <algorithm>               // NOLINT(build/include_order)
#include <cassert>                 // NOLINT(build/include_order)
#include <iostream>                // NOLINT(build/include_order)
#include <map>                     // NOLINT(build/include_order)
#include <set>                     // NOLINT(build/include_order)
#include <vector>                  // NOLINT(build/include_order)

inline void do_pass() {
/* pass */}

inline void inline_pass() {
/* pass */}

inline int indexing() {
  int sum = 0;
  std::vector<int> a = {};
  for (auto i : rangepp::xrange(10)) {
    a.push_back(i);
    sum += a[i];
  }
  return sum;
}

inline auto infer_bool(int code) {
  return ({
    std::vector<int> __tmp1 = {1, 2, 4};
    (std::find(__tmp1.begin(), __tmp1.end(), code) != __tmp1.end());
  });
}

inline void show() {
  int a1 = 10;
  int b9 = 2;
  int b2 = 2;
  assert(b2 == b9);
  double a2 = 2.1;
  std::cout << a2;
  std::cout << std::endl;
  for (auto i : rangepp::xrange(0, 10)) {
    std::cout << i;
    std::cout << std::endl;
  }
  for (auto i : rangepp::xrange(0, 10, 2)) {
    std::cout << i;
    std::cout << std::endl;
  }
  int a3 = -(a1);
  int a4 = a3 + a1;
  std::cout << a4;
  std::cout << std::endl;
  int sum1 = indexing();
  std::cout << sum1;
  std::cout << std::endl;
  std::vector<int> a5 = {1, 2, 3};
  std::cout << a5.size();
  std::cout << std::endl;
  std::vector<std::string> a9 = std::vector<std::string>{
      std::string{"a"}, std::string{"b"}, std::string{"c"}, std::string{"d"}};
  std::cout << a9.size();
  std::cout << std::endl;
  std::set<int> a6 = std::set<int>{1, 2, 3, 4};
  std::cout << a6.size();
  std::cout << std::endl;
  std::map<std::string, int> a7 =
      std::map<std::string, int>{{std::string{"a"}, 1}, {std::string{"b"}, 2}};
  std::cout << a7.size();
  std::cout << std::endl;
  bool a8 = true;
  if (a8) {
    std::cout << std::string{"true"};
    std::cout << std::endl;
  } else {
    if (a4 > 0) {
      std::cout << std::string{"never get here"};
      std::cout << std::endl;
    }
  }
  if (a1 == 11) {
    std::cout << std::string{"false"};
    std::cout << std::endl;
  } else {
    std::cout << std::string{"true"};
    std::cout << std::endl;
  }
  if (1 != NULL) {
    std::cout << std::string{"World is sane"};
    std::cout << std::endl;
  }
  do_pass();
  inline_pass();
  std::string s = std::string{"1    2"};
  std::cout << s;
  std::cout << std::endl;
}

int main(int argc, char **argv) {
  py14::sys::argv = std::vector<std::string>(argv, argv + argc);
  show();
}

File: tests/cases/coverage.py
from typing import List


def indexing():
    sum = 0
    a: List[int] = []
    for i in range(10):
        a.append(i)
        sum += a[i]
    return sum


def infer_bool(code: int):
    return code in [1, 2, 4]


def show():
    # assign
    a1 = 10
    # multi-assign
    b1 = b2 = 15
    assert b1 == 15
    assert b2 == 15
    b9 = 2
    b10 = 2
    assert b9 == b10
    # annotated assign
    a2: float = 2.1
    print(a2)
    # for loop
    print("range(0, 10")
    for i in range(0, 10):
        print(i)
    print("range(0, 10, 2")
    for i in range(0, 10, 2):
        print(i)
    print("range end")
    # for i in range(1, 10):
    #     print(i)
    # unary op
    a3 = -a1
    # binary op
    a4 = a3 + a1
    print(a4)
    # ternary op
    t1 = 10 if a1 > 5 else 5
    assert t1 == 10
    sum1 = indexing()
    print(sum1)
    # lists, sets and dict
    a5 = [1, 2, 3]
    print(len(a5))
    a9: List[str] = ["a", "b", "c", "d"]
    print(len(a9))
    a7 = {"a": 1, "b": 2}
    print(len(a7))
    a8 = True
    # print()
    if a8:
        print("true")
    elif a4 > 0:
        print("never get here")
    if a1 == 11:
        print("false")
    else:
        print("true")
    if 1 != None:
        print("World is sane")
    print(True)
    if True:
        a1 += 1
    assert a1 == 11
    if True:
        print("true")
    s = "1\
    2"
    print(s)
    assert infer_bool(1)
    # assert 1 != 2 != 3
    # Escape quotes
    _escape_quotes = """ foo "bar" baz """
    (_c1, _, _c2) = (1, 2, 3)


if __name__ == "__main__":
    show()
